### ISSUE TITLE
Delete should mark datastore elements as deleted but not delete them

### ISSUE THREAD

**Issue by wszarmach** [krico/jas#443]
Right now a delete operation will permanently remove from the datastore.
This makes it very difficult to investigate issues and can leave dangling references.
It would be better to mark these items as deleted but not actually delete them. Any query would then exclude deleted items.

**Comment by pauljanes** [comment]
Yes exactly. This one was raised before I believe. That cancelling from datastore may not be a good idea
